SEED-platform/seed
# Feat/derived columns editor

#### Any background context you want to provide?
SEED is adding ability for users to create custom columns derived from other column data.

#### What's this PR do?
Adds views and frontend for creating, editing, and deleting derived columns.

What this PR does NOT do:
We have discussed adding a "preview" of the derived column result for some inventory but that is NOT part of this PR. It will likely be added at a later time.

We have discussed adding a "warning" for unit aware columns. This is NOT included in this PR, I need to chat with someone more familiar with how to determine whether a column is unit aware. It will be added at a later time.

#### How should this be manually tested?
Go to your organization page and click the "Derived Columns" tab. You should be able to create, update, and delete derived column data. You should get helpful errors when things are invalid / incorrect.

Some notes:
- you should only be able to select columns for your currently active inventory type
- no duplicate parameter names or source columns are allowed
- parameter names must be a [valid Python identifier](https://docs.python.org/3/reference/lexical_analysis.html#identifiers)
- you must use all parameters in the expression
- you can't have undeclared parameters in the expression
- you must have at least one parameter (delete button should disappear)
- you can't edit the inventory type of an existing derived column
- the "create" or "save" button should generally be disabled if there are any active errors (except for those returned by the server, which are located below the expression and below the `Error(s):` title

#### What are the relevant tickets?
#1642 

#### Screenshots (if appropriate)
Editing existing column example
<img width="1081" alt="Screen Shot 2021-05-25 at 11 20 02 AM" src="https://user-images.githubusercontent.com/18518728/119524669-da717e80-bd4b-11eb-8eb3-5191fa91e1a1.png">

Example of different errors detected on frontend
<img width="1079" alt="Screen Shot 2021-05-25 at 11 22 50 AM" src="https://user-images.githubusercontent.com/18518728/119524576-c6c61800-bd4b-11eb-8d84-4338de1b2c83.png">

Example of errors detected on backend
<img width="1080" alt="Screen Shot 2021-05-25 at 11 23 38 AM" src="https://user-images.githubusercontent.com/18518728/119524627-d0e81680-bd4b-11eb-91d0-439813051698.png">
